src/status_im/navigation/state.cljs

```diff
@@ -2,6 +2,36 @@
 
 (defonce root-id (atom nil))
 (defonce pushed-screen-id (atom nil))
-(defonce curr-modal (atom nil))
 (defonce modals (atom []))
 (defonce dissmissing (atom false))
+(defonce navigation-state (atom []))
```

Comment:
why do we need this navigation state?

Reply:
don't we also have a navigation state for the jump-to screens?
Is it possible to just to share one state between the both if so?

Also is not easier to just have one atom exposed here with all of this data?

e.g
```
(defonce state (atom {
  :root-id nil
  :pushed-screen-id nil
  :modals []
  :dissmissing false
  :navigation-state [] })
```

Reply:
also the naming `navigation-state` is a bit strange here since the ns is basically that. What is the difference between this and the other states here?

Reply:
For navigation back logic. Otherwise we will know how to go back. (dismiss modal or pop or from which parent)

Reply:
additionally, rather than directly exposing the state we can expose a get method for these states?
mostly because we have other methods to update the state and this would help ensure that the state is being updated directly without using these methods.

Reply:
> > why do we need this navigation state?
> 
> For navigation back logic. Otherwise we will know how to go back. (dismiss modal or pop or from which parent)

so it's more like navigation-stack, or navigation-history?

Reply:
yes exactly